### PR TITLE
Assembler: Navigate to the sections screen with the category list

### PR DESCRIPTION
**Pull request by arthur791004**
Related to https://github.com/Automattic/wp-calypso/issues/81436

## Proposed Changes

* Add the Sections screen back. It shows the category list on the sidebar and patterns with the selected category on the panel next to the sidebar

**Screenshots**

image

**Demo**

https://github.com/Automattic/wp-calypso/assets/13596067/bacadfeb-853b-461c-9f8a-1dc8dd58f306

## Testing Instructions

* Go to Theme Showcase
* Select the Assembler CTA
* On the Assembler screen
  * Click “Sections”
  * Ensure the sidebar becomes the category list, and the first category is preselected
  * Click “< Back to Design your own” or “Save and continue”
  * Ensure the sidebar goes back to the previous one with “Header”, “Sections” and “Footer”

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [x] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [x] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [x] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [x] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/screen-sections.tsx
Comment (taipeicoder) [inline client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/screen-sections.tsx]: Is this string already translated? If not WDYT about using the previous one `Find the section patterns for your homepage by exploring the categories below.` and we can update the copy together with https://github.com/Automattic/wp-calypso/issues/81307?

Comment (arthur791004) [reply client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/screen-sections.tsx]: I thought it was and I'm wrong 🥲 Use `Find the section patterns for your homepage by exploring the categories below.` for non-English locale.

File: client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/screen-sections.tsx
Comment (taipeicoder) [inline client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/screen-sections.tsx]:
```suggestion
						'Find the right patterns for you by exploring the list of categories below.'
```

File: client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/screen-sections.tsx
Comment (taipeicoder) [inline client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/screen-sections.tsx]:
```suggestion
								'Find the right patterns for you by exploring the list of categories below.'
```

**Review by taipeicoder — COMMENTED** [review]
I noticed that the drawer in the Section panel doesn't close when you click the category twice, this is inconsistent with the drawer for Header/Footer 😕

What I mean is:
1. Click Sections.
2. The Blog Posts category is selected, with the drawer opened.
3. Clicking again the Blog Posts category doesn't close the drawer.

**Comment by arthur791004** [comment]
> I noticed that the drawer in the Section panel doesn't close when you click the category twice, this is inconsistent with the drawer for Header/Footer 😕

I thought it should work but I just found that the `onCategorySelect` won't be triggered when the category is active... I've resolved it 🙂

**Comment by fushar** [comment]
The back button on the Sections screen does not work as expected, when we previously clicked the "Save and continue" button. It somehow goes back to the last selected category. Recording:

https://github.com/Automattic/wp-calypso/assets/1525580/232c98be-8b2e-44ff-bbc2-04da003eef5e

File: client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/screen-main.tsx
Comment (fushar) [inline client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/screen-main.tsx:80]: We can now remove the `hasNestedItems` prop below (and its logic in `NavigatorItem`) :)

**Comment by arthur791004** [comment]
> The back button on the Sections screen does not work as expected, when we previously clicked the "Save and continue" button. It somehow goes back to the last selected category.

Nice catch! Fixed it via https://github.com/Automattic/wp-calypso/pull/81439/commits/1a272d1578afbcfec5cd3ae30e1a1dd3a9b1c008.

**Review by taipeicoder — APPROVED** [review]
Works as intended, thanks for expediting this 🎉 
Let's wait for @lucasmendes-design review before merging.

**Review by fushar — APPROVED** [review]
🔥

**Pull request merged**
